playwright/tests/page-objects/graph.ts

```diff
@@ -0,0 +1,56 @@
+import { expect, type Locator, type Page } from '@playwright/test';
+
+export class Graph {
+  private readonly page: Page;
+  private readonly graphRoot: Locator;
+
+  constructor(page: Page) {
+    this.page = page;
+    this.graphRoot = page.locator('.react-flow__viewport');
+  }
+
+  get nodes(): Locator {
+    return this.graphRoot.locator('.react-flow__node');
+  }
+
+  get edges(): Locator {
+    return this.page.locator('.react-flow__edge');
+  }
+
+  getNodeByText(text: string | RegExp): GraphNode {
+    const locator = this.nodes.filter({ hasText: text }).first();
+    return new GraphNode(locator);
+  }
+
+  async expectNodeCount(expectedCount: number) {
+    await expect(this.nodes).toHaveCount(expectedCount);
+  }
+
+  async expectEdgeCount(expected: number) {
+    await expect(this.edges).toHaveCount(expected);
+  }
```

Comment:
also this wrappers are IMO not needed. just call make this expect.tohavecount in the test.